Makefile

```diff
@@ -44,32 +63,54 @@ lint: ## Run golangci-lint
 verify: tidy generate format lint ## Verify the current code generation and lint
 	git diff --exit-code
 
-KUBERNETES_VERSION=v0.22.2
 COMBO_VERSION=$(shell git describe || echo $(DEFAULT_VERSION))
+BUILD_FLAGS=CGO_ENABLED=0
 VERSION_FLAGS=-ldflags "-X $(VERSION_PATH).GitCommit=$(GIT_COMMIT) -X $(VERSION_PATH).ComboVersion=$(COMBO_VERSION) -X $(VERSION_PATH).KubernetesVersion=$(KUBERNETES_VERSION)"
 build-cli: ## Build the CLI binary. Speciy VERSION_PATH, GIT_COMMIT, or KUBERNETES_VERSION to change the binary version.
-	$(Q)go build -a $(VERSION_FLAGS) -o ./bin/combo
+	$(Q) $(BUILD_FLAGS) go build $(VERSION_FLAGS) -o ./bin/combo
 
-IMAGE_REPO=quay.io/operator-framework/combo
-IMAGE_TAG=latest
 build-container: ## Build the Combo container. Accepts IMAGE_REPO and IMAGE_TAG overrides.
-	docker build . -f Dockerfile -t $(IMAGE_REPO):$(IMAGE_TAG)
+	docker build . -f Dockerfile -t $(IMAGE)
 
-CONTROLLER_GEN=$(Q)go run sigs.k8s.io/controller-tools/cmd/controller-gen
+build-local-container: build-cli ## Build the Combo container from the Dockerfile.local to speed compile time up. Accepts IMAGE_REPO and IMAGE_TAG overrides.
+	docker build . -f Dockerfile.local -t $(IMAGE)
 
-# Static tests.
-.PHONY: test test-unit verify build
+################
+# Test Targets #
+################
+.PHONY: test test-unit test-local test-e2e
 
-test: test-unit ## Run the tests
+test: test-unit test-e2e ## Run both the unit and e2e tests
 
+UNIT_TEST_DIRS=$(shell go list ./... | grep -v /test/)
 test-unit: ## Run the unit tests
-	$(Q)go test -count=1 -short ./...
+	$(Q)go test -count=1 -short $(UNIT_TEST_DIRS)
+
+test-e2e: ## Run the e2e tests
+	$(Q)go test -count=1 ./test/e2e
 
+###################
+# Running Targets #
+###################
+.PHONY: deploy teardown run run-local run-e2e run-e2e-local
 deploy: generate ## Deploy the Combo operator to the current cluster
 	kubectl apply --recursive -f manifests
 
 teardown: ## Teardown the Combo operator to the current cluster
 	kubectl delete --recursive -f manifests
 
-# Binary builds
-GO_BUILD := $(Q)go build
+IMAGE_LOAD_COMMAND=kind load docker-image
+run: build-container ## Run Combo on local cluster
+	$(IMAGE_LOAD_COMMAND) $(IMAGE)
+	$(MAKE) deploy
+
+LOCAL_CONTAINER_OS=linux
+LOCAL_CONTAINER_ARCH=amd64
+run-local: ## Run Combo on local environment with Dockerfile.local for faster deployment
+	$(MAKE) GOOS=$(LOCAL_CONTAINER_OS) GOARCH=$(LOCAL_CONTAINER_ARCH) build-local-container
```

Comment:
Shouldn't these be prereqs on `build-local-container` itself? Seems like it wouldn't work without them.

Reply:
So the reason that this is currently like this is because when we run locally we get around really length build times by building the binary first and then copying it onto the container. The problem with this is that we can't guarantee that the local OS matches that of the container. So line 110 calls that `build-local-container` target with the GOOS set to linux in order to force the local binary build to be runnable on Linux specifically. Would there be a better way to do this?

Reply:
Ah I see what you are saying now, yep I totally agree with this.